### PR TITLE
Expose "safe_serialization" parameter from AutoModel to HuggingFaceEm…

### PR DESCRIPTION
**Pull request by hkristof03**
…bedding

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Add support for [Nomic AI embeddings](https://github.com/run-llama/llama_index/issues/11935)

Fixes # (issue)

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def __init__(
         trust_remote_code: bool = False,
         device: Optional[str] = None,
         callback_manager: Optional[CallbackManager] = None,
+        safe_serialization: bool = False
```
Comment (hatianzhang) [inline llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py:71]: should it be optional?

Comment (hkristof03) [reply llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py]: You are right, I changed it.

**Comment by hkristof03** [comment]
@hatianzhang I am not able to figure out the issue,

**Comment by hatianzhang** [comment]
no worries I will fix it

**Comment by hkristof03** [comment]
> no worries I will fix it

Let's double check if this is correct first, based on this answer: https://github.com/run-llama/llama_index/issues/11935#issuecomment-1997844566

I didn't know that you have a dedicated module for nomic HF embeddings.

**Comment by hatianzhang** [comment]
@hkristof03 I think it should be fine to add this parameters for HF embedding in case other emb model needs it

**Comment by tomaarsen** [comment]
This has introduced a critical bug in `HuggingFaceEmbedding`:
```python
from llama_index.embeddings.huggingface import HuggingFaceEmbedding

embed_model = HuggingFaceEmbedding()
```

See more info in #12277. Please revert this PR - `AutoModel.from_pretrained` does not accept a `safe_serialization` parameter.

@logan-markewich @hatianzhang

**Comment by YanSte** [comment]
To solve:

 **Update > llama-index-embeddings-huggingface to 0.2.0**

**Comment by philiprhoades** [comment]
> To solve:
>
> Update > llama-index-embeddings-huggingface to 0.2.0

# pip install llama-index-embeddings-huggingface upgrade
Requirement already satisfied: llama-index-embeddings-huggingface in /usr/local/lib/python3.11/site-packages (0.1.5)
ERROR: Could not find a version that satisfies the requirement upgrade (from versions: none)
ERROR: No matching distribution found for upgrade

- I am using an old Fedora (38) so I can use Python3.11 - which I needed to do to fix an earlier problem . .

**Comment by Avinash-Raj** [comment]
@YanSte I can't since llama-index TEI has a version conflict.

```
llama-index-embeddings-text-embeddings-inference 0.1.2 requires llama-index-embeddings-huggingface<0.2.0,>=0.1.1, but you have llama-index-embeddings-huggingface 0.2.0 which is incompatible.
```

**Comment by riverind** [comment]
> llama-index-embeddings-huggingface

could you provide the difference between 0.2.0 and 0.1.x
since 
llama-index-embeddings-huggingface-optimum 0.1.5 requires llama-index-embeddings-huggingface<0.2.0,>=0.1.3, but you have llama-index-embeddings-huggingface 0.2.0 which is incompatible.

**Comment by logan-markewich** [comment]
Seems like just a dependency for optimum that wasn't bumped

0.2.x uses sentence transformers for the model.

**Comment by SuMiaoALi** [comment]
> > llama-index-embeddings-huggingface骆驼索引嵌入-huggingface
> 
> could you provide the difference between 0.2.0 and 0.1.x你能提供 0.2.0 和 0.1.x 之间的区别吗 since 自从 llama-index-embeddings-huggingface-optimum 0.1.5 requires llama-index-embeddings-huggingface<0.2.0,>=0.1.3, but you have llama-index-embeddings-huggingface 0.2.0 which is incompatible.llama-index-embeddings-huggingface-optimum 0.1.5 需要 llama-index-embeddings-huggingface<0 id=21>=0.1.3，但您有 llama-index-embeddings-huggingface 0.2.0，这是不兼容的。

Have you solve it now?

**Pull request merged**